### PR TITLE
Fibonacci

### PR DESCRIPTION
**Pull request by opiotrowsky**

### CODE REVIEW

**Comment by ziobron** [comment]
✅ CI Passed - Task: fibonacci
🏅 Granted 6 XP for @opiotrowsky

**Comment by ziobron** [comment]
⏱ 3 XP extra for @opiotrowsky for on-time delivery

File: homework/fibonacci/fibonacci.hpp [hunk 1]
```diff
@@ -1,11 +1,31 @@
 #pragma once
 
 int fibonacci_iterative(int sequence) {
-    // TODO: Your implementation goes here
-    return 0;
+    int first = 1, second = 1, third;
```
Comment (nauka-programowania-MA) [inline homework/fibonacci/fibonacci.hpp:4]: One definition/ declaration per line.

You should always initialize the variable created on stack with some default value.

File: homework/fibonacci/fibonacci.hpp [hunk 2]
```diff
@@ -1,11 +1,31 @@
...
+    
+    if(sequence == 0) {
+        return 0;
+    }
+    else if(sequence == 2 or sequence == 1) {
+        return first;
+    }
```
Comment (nauka-programowania-MA) [inline homework/fibonacci/fibonacci.hpp:11]: Sequence hold exactly the same value as you want to return for 0 and 1 :)

Better write:

`if (sequence < 2) {
    return sequence;
}`

You avoid second if

File: homework/fibonacci/fibonacci.hpp [hunk 3]
```diff
@@ -1,11 +1,31 @@
...
 
 int fibonacci_recursive(int sequence) {
-    // TODO: Your implementation goes here
-    return 0;
+    if(sequence == 0) {
+        return 0;
+    }
+    else if(sequence == 1 or sequence == 2) {
+        return 1;
+    }
```
Comment (nauka-programowania-MA) [inline homework/fibonacci/fibonacci.hpp:27]: Sequence hold exactly the same value as you want to return for 0 and 1 :)

Better write:

`if (sequence < 2) {
    return sequence;
}`

You avoid secodn if

File: homework/fibonacci/fibonacci.hpp [hunk 4]
```diff
@@ -1,11 +1,31 @@
...
+    else {
+        return fibonacci_recursive(sequence - 1) + fibonacci_recursive(sequence - 2);
+    }
```
Comment (nauka-programowania-MA) [inline homework/fibonacci/fibonacci.hpp:30]: If you return variable here, you don't need to use `else if` or `else`

**Review by nauka-programowania-MA — APPROVED** [review]

**Pull request closed without merging**